source/extensions/filters/network/thrift_proxy/header_transport_impl.cc

```diff
@@ -0,0 +1,299 @@
+#include "extensions/filters/network/thrift_proxy/header_transport_impl.h"
+
+#include <limits>
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/thrift_proxy/buffer_helper.h"
+#include "extensions/filters/network/thrift_proxy/transport_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+namespace {
+
+// c.f.
+// https://github.com/apache/thrift/blob/master/lib/cpp/src/thrift/protocol/TProtocolTypes.h#L27
+enum class HeaderProtocolType {
+  Binary = 0,
+  JSON = 1,
+  Compact = 2,
+
+  FirstHeaderProtocolType = Binary,
+  LastHeaderProtocolType = Compact,
+};
+
+} // namespace
+
+bool HeaderTransportImpl::decodeFrameStart(Buffer::Instance& buffer, MessageMetadata& metadata) {
+  if (buffer.length() < 14) {
```

Comment:
nit: should this be a constant (e.g.: `HEADER_LEN`)?

Reply:
Added some constants for the various sizes this transport uses.